service/provider/src/main/java/gov/va/vro/service/event/AuditEventProcessor.java

```diff
@@ -0,0 +1,60 @@
+package gov.va.vro.service.event;
+
+import gov.va.vro.camel.FunctionProcessor;
+import gov.va.vro.model.event.AuditEvent;
+import gov.va.vro.model.event.Auditable;
+import gov.va.vro.service.spi.audit.EventService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class AuditEventProcessor {
+
+  private final EventService eventService;
+
+  public Object logEvent(Object o, String routeId, String message) {
+    if (!(o instanceof Auditable auditableObject)) {
+      // object cannot be audited
+      return o;
+    }
+    AuditEvent event =
+        AuditEvent.builder()
+            .eventId(auditableObject.getEventId())
+            .payloadType(auditableObject.getClass())
```

Comment:
So we send the class and send the only interface value. Why not just send the object (auditableObject) itself? We will repeat the same lines in logException below.

Reply:
We do not log the object itself because it may contain PII. In the next ticket, I will provide a mechanism for selectively extract information from the object.